src/pack-n-test.ts

```diff
@@ -93,7 +111,7 @@ export async function packNTest(options: TestOptions) {
     const dependencies = sample.dependencies || [];
     const devDependencies = sample.devDependencies || [];
 
-    if (!isJSCodeSample(sample)) {
+    if (sample.ts) {
```

Comment:
I can't leave a comment on line 109, but it seems to me that we need to create some kind of branching behavior to install an ESM module, so instead of:

```
await execa('npm', ['init', '-y'], {cwd: installDir});
```

it would be:

```
await execa('npm', ['init', 'es6', '-y'], {cwd: installDir});
```

to make sure it works in an ESM package? I might be misunderstanding something though.

Reply:
The Node.js runtime has builtin support to ESM and there should be no need for much more boilerplate to get it 😊 

More specifically, the way this implementation targets `ESM` vs `CJS` is via file extension (`.mjs` vs `.cjs`) so there's no need for `{ "type": "module" }` which is the only bit that matters from what I see in that `npm init es6` boilerplate. ref: https://nodejs.org/dist/latest-v18.x/docs/api/packages.html#determining-module-system